### PR TITLE
Correct typos in Subjects and Subject Groups names

### PR DESCRIPTION
**Pull request by santiagorodriguez96**
There are probably still some typos to fix, but we can get continue to improve the list until we fix every one of them 🙂

### CODE REVIEW

File: lib/tasks/scrape.rake [hunk 1]
```diff
@@ -18,7 +20,7 @@ namespace :scrape do
     subject_groups.each do |_code, group|
       puts "Updating group #{group["code"]}"
       subject_group = SubjectGroup.find_or_initialize_by(code: group["code"])
-      subject_group.name = capitalize_name(group["name"])
+      subject_group.name = capitalize_name(correct_typos(group["name"].downcase))
```
Comment (santiagorodriguez96) [inline lib/tasks/scrape.rake:23]: This breaks subjects with roman numbers on their name so I'll have to change it

File: lib/typos_list.csv [hunk 2]
```diff
@@ -0,0 +1,22 @@
+administracion,administración
```
[the remark this answers is not in the document]
Comment (santiagorodriguez96) [reply lib/typos_list.csv]: [7c03de5](https://github.com/cedarcode/mi_carrera/pull/238/commits/7c03de571935cbf4c6a30d173d11333311c0b39e)

File: lib/tasks/scrape.rake (hunk 1 above)
Comment (santiagorodriguez96) [reply lib/tasks/scrape.rake]: Fixed in [f23d580](https://github.com/cedarcode/mi_carrera/pull/238/commits/f23d58043ecc8bef932e64963850242f6dda91ef) 🙂

File: lib/tasks/scrape.rake [hunk 3]
```diff
@@ -18,7 +18,7 @@ namespace :scrape do
     subject_groups.each do |_code, group|
       puts "Updating group #{group["code"]}"
       subject_group = SubjectGroup.find_or_initialize_by(code: group["code"])
-      subject_group.name = capitalize_name(group["name"])
+      subject_group.name = capitalize_name(correct_typos(group["name"]))
```
Comment (EmilioCristalli) [inline lib/tasks/scrape.rake:21]:
```suggestion
      subject_group.name = cleanup(group["name"])
```

wdyt about creating a function like `cleanup` or something like that, that does both the capitalization and typo correction? (also maybe we can avoid doing the `split` twice)

File: lib/tasks/scrape.rake [hunk 4]
```diff
@@ -64,6 +64,16 @@ def capitalize_name(name)
...
 end
 
+def correct_typos(text)
+  text.split.map do |word|
+    typos_list[word.downcase] || word
+  end.join(' ')
+end
+
+def typos_list
+  @typos_list ||= YAML.load_file(Rails.root.join("lib/typos_list.yml"))
```
Comment (EmilioCristalli) [inline lib/tasks/scrape.rake:74]: i think this is ok for now, but i'm a little worried about creating all these methods and instance variables in the top level scope.
wdyt about creating a ticket to refactor this code and move it to a module/class?

File: lib/tasks/scrape.rake (hunk 3 above)
Comment (santiagorodriguez96) [reply lib/tasks/scrape.rake:21]: I like what you are thinking 👀 

Maybe something like `format_name` instead?

File: lib/tasks/scrape.rake (hunk 4 above)
Comment (santiagorodriguez96) [reply lib/tasks/scrape.rake:74]: Yeah, I'm definitely up to it 👍

File: lib/tasks/scrape.rake (hunk 3 above)
Comment (EmilioCristalli) [reply lib/tasks/scrape.rake:21]: 👍  sounds good

**Review by EmilioCristalli — APPROVED** [review]

**Pull request merged**